sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/SchemaTranslation.java

```diff
@@ -240,4 +261,94 @@ private static FieldType fieldTypeFromProtoWithoutNullable(SchemaApi.FieldType p
             "Unexpected type_info: " + protoFieldType.getTypeInfoCase());
     }
   }
+
+  public static SchemaApi.Row rowToProto(Row row) {
```

Comment:
Same remark as before. The is another way of decoding a row to embed in a schema. I'm **wondering** if we can reuse the code that (de)serializes the row as an element? That was my first though in my Option prototype.

Reply:
No - I don't think we want to use SchemaCoder here. The advantage of serializing directly into the proto is that it makes the proto introspectable. Dumping binary blobs into the proto makes the proto much harder to use and to debug.

Reply:
OK, I'll as soon as the PR is merged in, I'll rebase the the option work and use .